tests/test_metadata.py

```diff
@@ -44,6 +45,31 @@ def test_pass(metadata):
     assert result.ret == 0
 
 
+def test_additional_metadata_from_json(testdir):
+    testdir.makepyfile(
+        """
+        def test_pass(metadata):
+            assert metadata.get('Imran') == 'Mumtaz'
+    """
+    )
+    result = testdir.runpytest("--metadata-from-json", '{"Imran": "Mumtaz"}')
+    assert result.ret == 0
```

Comment:
Case in point about not swallowing user made errors.

Say you remove the last double quote (after Mumtaz).

This test will fail (the assert will fail), but you won't know why.

Reply:
Makes sense. I can also add a negative test here to make sure we get a non 0 return code when giving bad json?

Reply:
> Makes sense. I can also add a negative test here to make sure we get a non 0 return code when giving bad json?

Well, that depends. If you add a custom message then sure. If not, adding a test would just be testing the `json` library (that it raises a ValueError).